system/Config/BaseConfig.php

```diff
@@ -90,15 +90,15 @@ public function __construct()
 		static::$moduleConfig = config('Modules');
 
 		$properties  = array_keys(get_object_vars($this));
-		$prefix      = get_class($this);
+		$prefix      = static::class;
 		$slashAt     = strrpos($prefix, '\\');
 		$shortPrefix = strtolower(substr($prefix, $slashAt === false ? 0 : $slashAt + 1));
 
 		foreach ($properties as $property)
 		{
 			$this->initEnvValue($this->$property, $property, $prefix, $shortPrefix);
 
-			if ($shortPrefix === 'encryption' && $property === 'key')
+			if ($this instanceof Encryption && $property === 'key')
```

Comment:
actually, parent should not know about the child implementation, but probably that next refactor, like have abstract method that called and overridden in child.